docs/src/index.md

```diff
@@ -0,0 +1,87 @@
+
+## Basic idea
+
+As in the example above, any MLJFlux model has a `builder` hyperparameter, an object encoding
+instructions for creating a neural network given the data that the
+model eventually sees (e.g., the number of classes in a classification
+problem). While each MLJ model has a simple default builder, users
+may need to define custom builders to get optimal results,
+and this will require familiarity with the [Flux
+API](https://fluxml.ai/Flux.jl/stable/) for defining a neural network
+chain.
+
+In the future MLJFlux may provide a larger assortment of canned
+builders. Pull requests introducing new ones are most welcome.
+
+## Flux or MLJFlux?
+[Flux](https://fluxml.ai/Flux.jl/stable/) is a deep learning framework in Julia that comes with everything you need to build deep learning models (i.e., GPU support, automatic differentiation, layers, activations, losses, optimizers, etc.). [MLJFlux](https://github.com/FluxML/MLJFlux.jl) wraps models built with Flux which provides a more high-level interface for building and training such models. More importantly, it empowers Flux models by extending their support to many common machine learning workflows that are possible via MLJ such as:
+
+- **Estimating performance** of your model using a holdout set or other resampling strategy (e.g., cross-validation) as measured by one or more metrics (e.g., loss functions) that may not have been used in training
+
+- **Optimizing hyper-parameters** such as a regularization parameter (e.g., dropout) or a width/height/nchannnels of convolution layer
+
+- **Compose with other models** such as introducing data pre-processing steps (e.g., missing data imputation) into a pipeline. It might make sense to include non-deep learning models in this pipeline. Other kinds of model composition could include blending predictions of a deep learner with some other kind of model (as in “model stacking”). Models composed with MLJ can be also tuned as a single unit.
+
+- **Controlling iteration** by adding an early stopping criterion based on an out-of-sample estimate of the loss, dynamically changing the learning rate (eg, cyclic learning rates), periodically save snapshots of the model, generate live plots of sample weights to judge training progress (as in tensor board)
+
+
+- **Comparing** your model with a non-deep learning models
+
+Thus, for model that could be implemented in both `Flux` and `MLJFlux`, one could choose working with `MLJFlux` instead of `Flux` if they are interested in any of the functionality above, while not willing to implement it from scratch and/or when they would prefer working with a more high-level interface equivalent to that of MLJ for their task.
```

Comment:
```suggestion
[`FastAI`](https://fluxml.ai/FastAI.jl/dev/FastAI@dev/doc/docs/introduction.md.html) also provides a high-level interface for interacting with `Flux` models. If you are working exclusively with neural network models, and are less interested with comparison/composition with other kinds of machine learning models, this may be more suitable choice. Otherwise, `MLJFlux` may be better, especially for the users already familiar with the MLJ interface.
```

Reply:
@ablaom while I have been only browsing and searching in the docs for the last few minutes, my conclusion so far is that `FastAI` does not go too much beyond `providing a high-level interface for interacting with Flux models. `

For instance, I could neither by searching their docs nor quickly browsing the pages find that they support features such as advanced out-of-sample evaluation (e.g., cross validation) or hyperparameter search (e.g., tuning the number of hidden layers/number of neurons per layers). [FluxTraining.jl](https://fluxml.ai/FluxTraining.jl/dev/documents/README.md) (related to FastAI in some way) seems to only support hyperparameter scheduling which is not exactly hyperparameter search (e.g., still can't tune the number of layers with this; can argue it's not really "tuning" whatsoever). That same package does support early stopping though.

Thus, I find the following to not the most accurate fact, in that `FastAI` is not a drop-in replacement for `MLJFlux`:

> If you are working exclusively with neural network models, and are less interested with comparison/composition with other kinds of machine learning models, this (i.e., FastAI) may be more suitable choice.

I'm also not sure why the mention `exclusively with neural network models` because in their README they tend to just speak more generically of deep learning models.


I would recommend one of the following line of actions:
- That we don't mention `FastAI` at all (i.e., what an average package would do; also avoids confusion to users that: (i), only know `MLJFlux` and  (ii), where `MLJFlux` is sufficient for them; if (ii) did not hold then they would have likely been able to find FastAI themselves)
- Go out of our way and maximize sincerity as you intended and mention what we know about `FastAI` (e.g., after the comparison with Flux we add a line like `A comparable project, FastAI/FluxTraining, also provides a high-level interface for interacting with Flux models and supports a subset of features that may overlap with (but not include all of) those supported by MLJFlux`

For the last point, one can also go into more detail in the comparison but I don't find this necessary as both packages are likely to change with time and users may be in general less interested in such comparison. Comparing `MLJFlux` and `Flux` makes lots of sense because it's guaranteed everyone likely knows `Flux` and this helps them understand `MLJFlux` better.

Reply:
I think it's helpful to make links to other related packages. I also admit, that (out of some laziness) I may have been too generous is stating what FastAI.jl can do. That said, I have had some interactions with the developers and indirectly with the project and have the impression it is a well thought-out design by experts, and I remember a lot of community engagement. If I were working exclusively with images today - and certainly video - I expect I might prefer FastAI to MLJFlux.  

So the second suggestion sounds good, which I agree is more accurate.

Reply:
Alright. I agree with everything you said. So we will go with replacing the text above with:
> A comparable project, FastAI/FluxTraining, also provides a high-level interface for interacting with Flux models and supports a set of features that may overlap with (but not include all of) those supported by MLJFlux.

Right? Let me know if you would to fine-tune it further.

Reply:
all good